Decode/PerfItemValue.cs

```diff
@@ -10,58 +10,115 @@ namespace Microsoft.LinuxTracepoints.Decode
+    /// <br/>
+    /// Array field (array-begin or array-end item).
+    /// Array element type is simple (fixed-size element).
+    /// <br/>
+    /// <c>ElementCount</c> is the number of elements in the array.
+    /// <br/>
+    /// <c>Format</c> is significant and <c>StructFieldCount</c> should be ignored
+    /// (simple type is never <c>Struct</c>).
+    /// <br/>
+    /// For array-end, <c>Bytes</c> will be empty.
+    /// <br/>
+    /// For array-begin, <c>Bytes</c> contains the field's values and
+    /// <c>Bytes.Length == TypeSize * ElementCount</c>,
+    /// e.g. for a <c>Value32</c>, <c>TypeSize == 4</c> and <c>Bytes.Length == 4 * ElementCount</c>.
+    /// You may use <see cref="EventHeaderEnumerator"/><c>.MoveNext()</c> to visit the array elements,
+    /// or you may process the array values directly and then use
+    /// <see cref="EventHeaderEnumerator"/><c>.MoveNextSibling()</c> to skip the array elements.
     /// </item>
+    ///
     /// <item>
-    /// The end of an array of simple type:
-    /// Bytes is empty.
-    /// Type contains type information for the array.
-    /// </item>
-    /// <item>
-    /// The beginning or end of an array of complex elements:
-    /// Bytes is empty (array elements are accessed via the event enumerator).
-    /// Type contains type information for the array.
+    /// <b>Complex array:</b> <c>!IsScalar &amp;&amp; TypeSize == 0</c>
+    /// <br/>
+    /// Array field (array-begin or array-end item).
+    /// Array element type is complex (variable-size or struct element).
+    /// <br/>
+    /// <c>ElementCount</c> is the number of elements in the array.
+    /// <br/>
+    /// If <c>Encoding == Struct</c>, this is the beginning or end of an array of structures,
+    /// <c>Format</c> should be ignored, and <c>StructFieldCount</c> is significant.
+    /// Otherwise, this is an array of variable-length values, <c>Format</c> is significant,
+    /// and <c>StructFieldCount</c> should be ignored.
+    /// <br/>
+    /// <c>Bytes</c> will be empty. Use <see cref="EventHeaderEnumerator"/><c>.MoveNext()</c>
+    /// to visit the array elements.
     /// </item>
+    ///
     /// </list>
     /// </summary>
     public readonly ref struct PerfItemValue
     {
         /// <summary>
-        /// Initializes a new instance of the PerfItemValue struct. These are normally created
-        /// by EventHeaderEnumerator.GetItemInfo() or by PerfFieldFormat.GetFieldValue().
+        /// Initializes a new instance of the PerfItemValue struct.
+        /// These are not normally created directly. You'll normally get instances of this struct from
+        /// <see cref="EventHeaderEnumerator"/><c>.GetItemInfo()</c> or
+        /// <see cref="PerfFieldFormat"/><c>.GetFieldValue()</c>.
         /// </summary>
         public PerfItemValue(
             ReadOnlySpan<byte> bytes,
-            PerfItemType type)
+            PerfItemMetadata metadata)
```

Comment:
Note: Most of the changes in this file are just from the rename of `type` to `metadata`.